nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -223,7 +229,7 @@ class BtActionNode : public BT::ActionNodeBase
         feedback_.reset();
 
         auto goal_status = goal_handle_->get_status();
-        if (goal_updated_ && (goal_status == action_msgs::msg::GoalStatus::STATUS_EXECUTING ||
+        if (goal_updated_ && should_send_goal_ && (goal_status == action_msgs::msg::GoalStatus::STATUS_EXECUTING ||
```

Comment:
I think this is for preemption which we'd always want to run? The `on_wait_for_result()` function may choose or not to update `goal_updated_` so I think that would meet the need here. I wouldn't think someone would set `goal_updated_` to true while also setting `should_send_goal_` to false after ticking the node. That feels like they meet the same intention just in the initial tick vs preemption during execution contexts